File: leetcode/Easy/9. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        } else if (x >= 0 && x < 10) {
            return true;
        }
        
        long long rev = 0, tmp = x;
        
        while (tmp != 0) {
            int num = tmp % 10;
            rev = rev*10 + num;
            tmp /= 10;
        }
        
        if (rev == x) {
            return true;
        } else {
            return false;   
        }
    }
};

File: leetcode/Easy/9. Palindrome Number.py
class Solution:   
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x > 0 and x % 10 == 0):
            return False
        
        return x == self.reverseNumber(x)
    
    def reverseNumber(self, x: int) -> int:
        num = 0
        
        while x > 0:
            num = num*10 + x%10
            x //= 10
            
        return num